certbot/certbot/_internal/client.py

```diff
@@ -208,11 +205,22 @@ def register(config: configuration.NamespaceConfig, account_storage: AccountStor
         config.email = None
 
     # Each new registration shall use a fresh new key
-    rsa_key = generate_private_key(
+    key: jose.JWK
+    if not config.ecdsa_account_key:
+        rsa_key = rsa.generate_private_key(
             public_exponent=65537,
             key_size=config.rsa_key_size,
             backend=default_backend())
-    key = jose.JWKRSA(key=jose.ComparableRSAKey(rsa_key))
+        key = jose.JWKRSA(key=jose.ComparableRSAKey(rsa_key))
+    else:
+        # RFC 8555 6.2 says only ES256 is supported, i.e., SECP256R1
+        # mypy complains without the cast, although
+        # EllipticCurvePrivateKeyWithSerialization = EllipticCurvePrivateKey
```

Comment:
Hm I'm not sure either because these are indeed the same type.

Ideally we'd continue to investigate this, but I think a cast with a link to `cryptography` code declaring them as the same type is fine too.